components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/ChallengeQuestionManager.java

```diff
@@ -550,12 +551,18 @@ public ChallengeQuestion getUserChallengeQuestion(User user, String challengesUr
     private Map<String, String> retrieveAnsweredChallenges(User user, UserChallengeAnswer[] userChallengeAnswers)
             throws IdentityRecoveryException {
 
-        List<String> claimsList = new ArrayList<>();
-        for (UserChallengeAnswer answer : userChallengeAnswers) {
-            claimsList.add(answer.getQuestion().getQuestionSetId().trim());
+        Map<String, String> existingQuestionAndAnswers;
+        if (!ArrayUtils.isEmpty(userChallengeAnswers)) {
```

Comment:
ArrayUtils has a isNotEmpty method. It can be directly used

Reply:
Fixed 4a0c990